include/rapidjson/internal/dtoa.h

```diff
@@ -41,7 +41,7 @@ inline void GrisuRound(char* buffer, int len, uint64_t delta, uint64_t rest, uin
     }
 }
 
-inline unsigned CountDecimalDigit32(uint32_t n) {
+inline int CountDecimalDigit32(uint32_t n) {
```

Comment:
The point is that signed integer types should almost always be the choice whenever an arithmetic operation (+, -, *, /, etc) applys on that object, regardless whether the range of the value may be negative or not, because the wrapping behavior of unsigned integer types are very hard for the tools (such as UBsan) to reason your program, and often causes the compilers to produce downgraded code.  The YouTube video I pasted above shows a typical problem: using 32-bit unsigned integer in array offset on 64-bit architecture causes the compiler to "simulate" the wrapping behavior to ensure the address not overflowing the 64-bit address space, by changing the offset to 32-bit signed integer then the program is transformed into using only one 64-bit register because overflowing signed integer type itself is UB.  In summary, using `int` as raw array index is just fine, using `unsigned` may be bad for architectures with longer word.

Going back to the return value here, it's later being used to access array, very similar to the example above, so there is nothing wrong to changed it to signed.